packages/core/src/install/index.ts

```diff
@@ -228,11 +232,19 @@ export async function mutateModules (
     let needsFullResolution = !maybeOpts.ignorePackageManifest && (
       !equals(ctx.wantedLockfile.overrides ?? {}, overrides ?? {}) ||
       !equals((ctx.wantedLockfile.neverBuiltDependencies ?? []).sort(), (neverBuiltDependencies ?? []).sort()) ||
+      // internally we use false to indicate: this lockfile does not have a onlyBuiltDependencies key
+      // if it is an empty array, it means no packages is allowed to be built.
```

Comment:
an empty array may mean just that `ignore-scripts` is true

Reply:
this is the comment for the new options I added.